src/lib/tripAlgorithm.ts

```diff
@@ -1,99 +1,72 @@
-
-    //массив времени пути от начальной точки до остальных организаций (т.к. проверяем, успеем ли вернуться домой) -- пока что берется из данных (старт->все организации) -- нужно ли отдельно считать в обратном направлении?
-    let IsStart = true;
-    let routeIsBuilt = false;
-    let backwardTime = [];
-
-    _getNextRoute(currentPoint, visitedOrganizations, unvisitedOrganizations, currentTime, ultimateTime, IsStart, routeIsBuilt, backwardTime, callback);
+    getNextRoute(currentPoint, visitedOrganizations, unvisitedOrganizations, currentTime, ultimateTime, routeIsBuilt, callback);
 }
 
 
 //функция, возвр массив [visit, unvisit] на одном шаге алгоритма
-function _getNextRoute(currentPoint, visitedOrganizations, unvisitedOrganizations, currentTime, ultimateTime, IsStart, routeIsBuilt, backwardTime, callback){
-    _getSortedRoutesTimeArray(currentPoint, unvisitedOrganizations, function(routesTime){
-        if (IsStart) {
-            backwardTime = routesTime;
-            IsStart = false;
-        } 
-        let visitIsPossible = false;    //сможем ли посетить организацию
+function getNextRoute(currentPoint: IAlgorithmOrganizationParam, visitedOrganizations: Array<IAlgorithmOrganizationParam>, 
+        unvisitedOrganizations: Array<IAlgorithmOrganizationParam>, currentTime: number, ultimateTime: number, 
+        routeIsBuilt: boolean, callback: any){
+    getSortedRoutesTimeArray(currentPoint, unvisitedOrganizations, function(routesTime: Array<IOrgTravelTime>){
+        let visitIsPossible: boolean = false;    //сможем ли посетить организацию
         //проверить, успеваем ли в ближайшую организацию
         for (let rt of routesTime){
-            if (visitIsPossible) break;
-            let arrivalTime = currentTime + rt.travelTime;  //время прибытия в организацию = текущее время дня + время чтоб добраться до организации
-            let departureTime = arrivalTime + rt.organization.timeSpend;    //время отбытия из организации = время прибытия + время посещения
+            if (visitIsPossible) break;    
             /*можем посетить, если:
-            departureTime <= время завершения работы организации
-            && departureTime + route(организация, начальная точка) <= время завершения дневной экскурсии, т.е. успеваем вернуться домой*/
-            let backTime = _findTimeById(rt, backwardTime);  //время для возвращения в начальную точку
+            приехали допустимо раньше открытия или обеда (тогда departureTime будет меняться по-разному)
+            уедем раньше начала обеда или закрытия
+            успеем домой    
+           */
+            let backTime: number = rt.backwardTime;  //время для возвращения в начальную точку
+            let arrivalTime: number = currentTime + rt.travelTime;  //время прибытия в организацию = текущее время дня + время чтоб добраться до организации
+                
+            const maxWaitingTime: number = 10; //максимально допустимое время ожидания перед открытием
```

Comment:
про 10 минут я условно сказал. Вообще, конечно, если это последняя точка и время ожидания = 10 часов, значит нужно ждать 10 часов. Если ты тут напишешь 24 часа, это сделает алгоритм неоптимальным?

```suggestion
            const maxWaitingTime: number = 24 * 60; //максимально допустимое время ожидания перед открытием
```

Reply:
По поводу 24 часов -- да, мы же будем просто ждать и не посетим остальные места. Можно сделать как раньше -- проверять, что мы просто успеваем посетить организацию до закрытия и убрать эти 10 минут. Или дополнительное условие с последней точкой придумывать, но можно не успеть.

Reply:
Ну у тебя же жадный алгоритм. Т.е. ты должна точку, где нужно подождать, сравнить с другими точками и понять, стоит ли столько ждать или лучше поехать в другое место, а в это потом вернуться. Короче, этой переменной вообще не должно быть

Reply:
Тогда как такой вариант: если приехали до открытия и если из непросмотренных организаций не существует такой, в которую приедем раньше, чем в текущую, то включаем в маршрут текущую организацию и ждем открытия?

Reply:
Да, хороший